finality-aleph/src/network/service.rs

```diff
@@ -145,24 +173,44 @@ impl<N: Network, D: Data> Service<N, D> {
             }
             Event::NotificationStreamOpened {
                 remote, protocol, ..
-            } => {
-                if protocol == ALEPH_PROTOCOL_NAME {
+            } => match protocol.as_ref().try_into() {
+                Ok(Protocol::Generic) => {
                     trace!(target: "aleph-network", "NotificationStreamOpened event for peer {:?}", remote);
                     let (tx, rx) = mpsc::channel(PEER_BUFFER_SIZE);
                     self.spawn_handle.spawn(
                         "aleph/network/peer_sender",
                         None,
                         self.peer_sender(remote.into(), rx),
                     );
-                    self.connected_peers.insert(remote.into());
-                    self.peer_senders.insert(remote.into(), tx);
+                    self.generic_connected_peers.insert(remote.into());
+                    self.generic_peer_senders.insert(remote.into(), tx);
                 }
-            }
+                Ok(Protocol::Validator) => {
+                    trace!(target: "aleph-network", "NotificationStreamOpened event for peer {:?}", remote);
+                    let (tx, rx) = mpsc::channel(PEER_BUFFER_SIZE);
+                    self.spawn_handle.spawn(
+                        "aleph/network/peer_sender",
+                        None,
+                        self.peer_sender(remote.into(), rx),
+                    );
+                    self.validator_connected_peers.insert(remote.into());
+                    self.validator_peer_senders.insert(remote.into(), tx);
+                }
+                Err(_) => {}
```

Comment:
We don't need to at least log in that situations right (to avoid silent errors); or are those really unreachable paths